applications/GeoMechanicsApplication/tests/test_helper.py

```diff
@@ -419,6 +420,97 @@ def find_closest_index_greater_than_value(input_list, value):
     return None
 
 
+def are_values_almost_equal(expected: Any, actual: Any, abs_tolerance: float = 1e-7) -> bool:
+    """
+    Checks whether two values are almost equal.
+
+    Args:
+        - expected (Any): Expected value.
+        - actual (Any): Actual value.
+
+    Returns:
+        - True if the values are almost equal, False otherwise.
+
+    """
+    # check if the value is a dictionary and check the dictionary
+    if isinstance(expected, dict):
+        # check if the value is a dictionary and check the dictionary
+        if not are_dictionaries_almost_equal(expected, actual):
+            return False
+    elif isinstance(expected, str):
+        # check if the value is a string and compare the strings
+        if expected != actual:
+            return False
+    elif isinstance(expected, (list, tuple, set)):
+        # check if the value is a list, tuple or set and compare the values
+        if not are_iterables_almost_equal(expected, actual):
+            return False
+    elif expected is None:
+        # check if the value is None and compare the values
+        if actual is not None:
+            return False
+    elif isinstance(expected, (float, int, complex)):
+        # The value is a number and compare the values
+        if not math.isclose(expected, actual, abs_tol=abs_tolerance):
+            return False
+    else:
+        raise Exception(f"Unsupported type {type(expected)}")
+
+    return True
+
+
+def are_iterables_almost_equal(expected: (list, tuple, set), actual: (list, tuple, set),
+                               abs_tolerance: float = 1e-7) -> bool:
+    """
+    Checks whether two iterables are almost equal.
+
+    Args:
+        - expected (list, tuple, set): Expected iterable.
+        - actual (list, tuple, set): Actual iterable.
+
+    Returns:
+        - True if the iterables are almost equal, False otherwise.
+
+    """
+    # check if the value is a list, tuple or set and compare the values
+    if len(expected) != len(actual):
+        return False
+
+    for v_i, actual_i in zip(expected, actual):
+        if not are_values_almost_equal(v_i, actual_i, abs_tolerance):
+            return False
+
+    return True
```

Comment:
I would suggest to use the built-in function [`all`](https://docs.python.org/3/library/functions.html#all) to get rid of the explicit `for` loop:
```suggestion
    return all([are_values_almost_equal(expected_value, actual_value, abs_tolerance) for expected_value, actual_value in zip(expected, actual)])
```

Reply:
I think this is less efficient, as in your version, the for loop is still there within the list comprehension.

But in your case, the loop does not exit early if values are not almost equal